pkg/writer/junit_xml_test.go

```diff
@@ -0,0 +1,167 @@
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package writer
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/accurics/terrascan/pkg/policy"
+	"github.com/accurics/terrascan/pkg/results"
+)
+
+func TestJUnitXMLWriter(t *testing.T) {
+	testOutput := `
+<testsuites tests="566" name="TERRASCAN_POLICY_SUITES" failures="1" time="0">
+  <testsuite tests="566" failures="1" time="0" name="TERRASCAN_POLICY_SUITE" package="test">
+    <properties>
+      <property name="Terrascan Version" value="v1.3.1"></property>
+    </properties>
+    <testcase classname="modules/m1/main.tf" name="[ERROR] resource: &#34;bucket&#34; at line: 20 violates: AWS.S3Bucket.DS.High.1043" severity="HIGH" category="S3">
+      <failure message="Description: S3 bucket Access is allowed to all AWS Account Users., File: modules/m1/main.tf, Line: 20, Severity: HIGH, Rule Name: s3EnforceUserACL, Rule ID: AWS.S3Bucket.DS.High.1043, Resource Name: bucket, Resource Type: aws_s3_bucket, Category: S3" type=""></failure>
+    </testcase>
+    <testcase classname="modules/m1/main.tf" name="[ERROR] resource: &#34;bucket&#34; at line: 20 violates: AWS.S3Bucket.DS.High.1043" severity="HIGH" category="S3">
+      <skipped message=""></skipped>
+      <failure message="Description: S3 bucket Access is allowed to all AWS Account Users., File: modules/m1/main.tf, Line: 20, Severity: HIGH, Rule Name: s3EnforceUserACL, Rule ID: AWS.S3Bucket.DS.High.1043, Resource Name: bucket, Resource Type: aws_s3_bucket, Category: S3" type=""></failure>
+    </testcase>
+  </testsuite>
+</testsuites>
+  `
+	testOutputNoViolations := `
+<testsuites tests="550" name="TERRASCAN_POLICY_SUITES" failures="1" time="0">
+  <testsuite tests="550" failures="1" time="0" name="TERRASCAN_POLICY_SUITE" package="test_resource_path">
+    <properties>
+      <property name="Terrascan Version" value="v1.3.1"></property>
+    </properties>
+  </testsuite>
+</testsuites>
+	`
+
+	type args struct {
+		data interface{}
+	}
+	tests := []struct {
+		name       string
+		args       args
+		wantWriter string
+		wantErr    bool
+	}{
+		{
+			name: "incorrect input for JunitXMLWriter",
+			args: args{
+				// some invalid data
+				data: struct{ Name string }{Name: "test"},
+			},
+			wantErr: true,
+		},
+		{
+			name: "data with violations and skipped violations",
+			args: args{
+				data: violationsInput,
+			},
+			wantWriter: testOutput,
+		},
+		{
+			name: "data with no violations",
+			args: args{
+				policy.EngineOutput{
+					ViolationStore: &results.ViolationStore{
+						Summary: results.ScanSummary{
+							ResourcePath:     "test_resource_path",
+							IacType:          "k8s",
+							Timestamp:        "2020-12-12 11:21:29.902796 +0000 UTC",
```

Comment:
@patilpankaj212 you sure about hardcoding the timestamp?

Reply:
Is there anything wrong with it?